### PR TITLE
Piston control

### PR DESCRIPTION
**Pull request by RoboCD**
Adds piston control to warehouse robot and an example piston to the articubot

### CODE REVIEW

File: ros2_ws/src/warehouse_robot/config/my_controllers.yaml [hunk 1]
```diff
@@ -9,6 +9,20 @@ controller_manager:
...
+    # joint_piston:
+      # type: position_controllers/JointGroupPositionController
+
+    joint_piston:
+      type: forward_command_controller/ForwardCommandController
+
+  # joint1_position_controller:
+  #   type: effort_controllers/JointPositionController
+  #   joint: joint_piston
+  #   pid: {p: 1000, i: 0, d: 0}
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/config/my_controllers.yaml]: why are these and some of the lines above commented out?  should they just be removed instead?  If not, a quick comment detailing why they're commented out but not removed would be valuable.

File: ros2_ws/src/warehouse_robot/config/my_controllers.yaml [hunk 2]
```diff
@@ -65,4 +79,23 @@ diff_cont:
...
\ No newline at end of file
+#   ros__parameters:
+
+
+# joint_piston:
+#   ros__parameters:
+
+#     joints:
+#       - piston_joint
+#     interface_name: position
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/config/my_controllers.yaml]: same note about commented out lines from above.

File: ros2_ws/src/warehouse_robot/description/robot_core.xacro [hunk 3]
```diff
@@ -184,4 +192,72 @@
...
+
+    <gazebo reference="piston">
+        <material>Gazebo/White</material>
+        <selfCollide>true</selfCollide>
+        <gravity>0</gravity>
+        <!-- <mu1 value="0.1" />
+        <mu2 value="0.1" />
+        <kp value="10000000.0" />
+        <kd value="10000" />
+        <fdir1 value="0 0 0" /> -->
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/description/robot_core.xacro]: should these be kept or removed?  Looks like it's for a PD controller?

File: ros2_ws/src/warehouse_robot/description/robot_core.xacro [hunk 4]
```diff
@@ -184,4 +192,72 @@
...
+        </collision>
+        <xacro:inertial_cylinder mass="${tray_mass}" length="${tray_height}" radius="${piston_radius}">
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+        </xacro:inertial_cylinder>
+    </link>
+
+    <gazebo reference="caster_wheel">
+        <material>Gazebo/White</material>
+        <!-- <mu1 value="0.001"/>
+        <mu2 value="0.001"/> -->
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/description/robot_core.xacro]: should i stay or should i go?  Comment on why if I stay.

File: ros2_ws/src/warehouse_robot/launch/launch_sim.launch.py [hunk 5]
```diff
@@ -94,5 +105,7 @@ def generate_launch_description():
         gazebo,
         spawn_entity,
         diff_drive_spawner,
-        joint_broad_spawner
+        joint_broad_spawner,
+        joint_piston_spawner,
+        # joint_state_publisher_gui_node
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/launch/launch_sim.launch.py]: stay or go?

File: ros2_ws/src/warehouse_robot/launch/rsp.launch.py [hunk 6]
```diff
@@ -19,9 +19,12 @@ def generate_launch_description():
     # Process the URDF file
     pkg_path = os.path.join(get_package_share_directory('warehouse_robot'))
     xacro_file = os.path.join(pkg_path,'description','robot.urdf.xacro')
+    # Uncomment this line to switch to the warehouse robot
+    # xacro_file = os.path.join(pkg_path,'urdf','robot.urdf.xacro')
```
Comment (zekeflaton) [inline ros2_ws/src/warehouse_robot/launch/rsp.launch.py:23]: nice comment for commented out lines :)  exactly what imlooking for above.

**Review by zekeflaton — COMMENTED** [review]
couple of nits on commenting or cleaning up the code.  Note, the comments I'm talking about should be in the file, not in a reply on github.

File: ros2_ws/src/warehouse_robot/config/my_controllers.yaml (hunk 1 above)
Comment (RoboCD) [reply ros2_ws/src/warehouse_robot/config/my_controllers.yaml]: That was the original from the `warehouse_robot` and I tried to closely match it. It's not needed here though, I'll remove it

File: ros2_ws/src/python_controllers/setup.py [hunk 7]
```diff
@@ -16,7 +16,7 @@
 setup(
     name=package_name,
     version='0.0.0',
-    packages=[package_name, package_name+'/scripts'],
+    packages=[package_name, package_name+'/scripts', package_name+'/src'],
```
Comment (zekeflaton) [inline ros2_ws/src/python_controllers/setup.py:19]: Thanks

**Review by zekeflaton — APPROVED** [review]

**Pull request merged**